### ISSUE TITLE
relative positioning for access rules fails for rulebases with more than 50 rules

### ISSUE THREAD

**Issue by 2ps** [CheckPointSW/CheckPointAnsibleMgmtCollection#69]
Using relative positioning in a rulebase will trigger the use of `get_number_from_position`.  However, the `get_number_from_position` function does not paginate the entire rulebase and, instead, only looks at the first 50 rules.  This poses a problem for those of us trying to use relative positioning for determining the position of a rule in a rulebase since if the rule relative to which we are positioning is not in the first 50, the rule will not be positioned correctly.

**Comment by 2ps** [comment]
Suggested code change:

```
def get_rulebase(connection, version, layer):
    offset = 0
    while True:
        payload_for_show_access_rulebase = {
            'name': layer,
            'limit': 200,
            'offset': offset,
        }
        code, response = send_request(connection, version, 'show-access-rulebase', payload_for_show_access_rulebase)
        offset = response['to']
        total = response['total']
        rulebase = response['rulebase']
        for rules in rulebase:
            if 'rulebase' in rules:
                yield from rules['rulebase']
            yield rules
        if total <= offset:
            return


def get_below_position(rule):
    if rule['type'] == 'access-rule':
        position = int(rule['rule-number'])
        return position + 1
    if rule['type'] == 'access-section':
        # else we have an access section
        position = int(rule['from'])
        return position
    return 1


def get_above_position(rule):
    if rule['type'] == 'access-rule':
        position = int(rule['rule-number'])
        return max(position - 1, 1)
    if rule['type'] == 'access-section':
        # else we have an access section
        position = int(rule['from'])
        return max(position - 1, 1)
    return 1


# get the position in integer format
def get_number_from_position(payload, connection, version):
    if 'position' in payload:
        position = payload['position']
        if type(payload['position']) is not dict:
            return int(position)
        layer = payload['layer']
        for rule in get_rulebase(connection, version, layer):
            rule_name = rule.get('name') or rule['uid']
            below_rule = position.get('below')
            above_rule = position.get('above')
            if not below_rule and not above_rule :
                return None
            if rule_name == below_rule:
                return get_below_position(rule)
            if rule_name == above_rule:
                return get_above_position(rule)
    else:
        return None

    # This code relevant if we will decide to support 'top' and 'bottom' in position

    # position_number = None
    # # if position is not int, convert it to int. There are several cases: "top"
    # if position == 'top':
    #     position_number = 1
    # elif position == 'bottom':
    #     payload_for_show_access_rulebase = {'name': payload['layer'], 'limit': 0}
    #     code, response = send_request(connection, version, 'show-access-rulebase', payload_for_show_access_rulebase)
    #     position_number = response['total']
    # elif isinstance(position, str):
    #     # here position is a number in format str (e.g. "5" and not 5)
    #     position_number = int(position)
    # else:
    #     # here position suppose to be int
    #     position_number = position
    #
    # return position_number
```

**Comment by chkp-edenbr** [comment]
Hi @2ps 

Issue fixed in new release v4.0.0.

Regards,
Eden
